### PR TITLE
API key generation

### PR DESCRIPTION
**Pull request by gentlecat**
New table and UI.

### CODE REVIEW

File: webserver/static/scripts/profile.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+$(document).ready(function () {
+    let apiKey = $("#api-key");
+    if (apiKey) {  // Only when key is displayed (user is viewing their own page)
+
+        // Not showing confirmation dialog if there's no active key (key value is not displayed).
+        let ignoreConfirmation = apiKey.css("display") === "none";
+
+        $("#btn-generate-api-key").click(function () {
+            if (ignoreConfirmation || confirm('Are you sure you want to generate new API key?\n' +
```
Comment (alastair) [inline webserver/static/scripts/profile.js]: I really don't like javascript popups. Can we do it with new pages or javascript to change the contents of the page?

**Comment by alastair** [comment]
What's the purpose of API keys? Who will be using them?
 * Website frontend
 * Other apps who need to read/write user information

Some questions:
 * Is it OK for a user to have only 1 key?
 * Do all keys have the same permissions? r/w?
 * Should we just take the plunge and put oauth in like we have for CB?

File: webserver/static/scripts/profile.js (hunk 1 above)
Comment (gentlecat) [inline webserver/static/scripts/profile.js]: Yeah, I guess we could, but that's more work which I'm not sure is necessary for what we are doing. What's wrong with popups?

**Comment by gentlecat** [comment]
Our website will keep using a cookie for authentication.

> Other apps who need to read/write user information

Just datasets for now.

> Is it OK for a user to have only 1 key?

There might be a need for multiple keys if you have several apps and don't want to reuse the same key. I think for now one should be enough. In future, if needed, it will be straightforward to add support for multiple.

File: webserver/static/scripts/profile.js (hunk 1 above)
Comment (gentlecat) [inline webserver/static/scripts/profile.js]: Can also remove it completely and just send the request.

**Pull request merged**